File: HMWK_02_tvt3804/hmwk_02.py
# Tran, Thang
# 1001-233-804
# 2018-03-10
#--------------------------------------------------
import re
import sys

#--------------------------------------------------
def processToken( token ) :

    isID = r"^[a-mN-Z_][n-zA-M0-9_]*$"
    isInt = r"^(0d|0D)[0-9]+$"
    isnotEFP = r"^[0-9]+$"
    isHexInt = r"^(0x|0X)[0-9a-fA-F]+$"
    isFP = r"^[0-9]+[\.][0-9]+$"
    isEFP = r"(([0-9]+\.)|(\.[0-9]+)|([0-9]+\.[0-9]+)|([0-9]+))([eE][-+]?[0-9]+)?$"

    if re.match(isID, token):
        print('>%s<  matches ID.' % token)
    elif re.match(isInt, token):
        print('>%s<  matches INT.' % token)
    elif re.match(isnotEFP,token):
         print('>%s<  does not match.' % token)
    elif re.match(isHexInt,token):
        print('>%s<  matches HEXINT.' % token)
    elif re.match(isFP, token):
        print('>%s<  matches FP.' % token)
    elif re.match(isEFP,token):
        print('>%s<  matches EFP.' % token)
    else:
        print('>%s<  does not match.' % token)

#--------------------------------------------------
def main() :
  fName = sys.argv[1]
  print( 'processing tokens from ' + fName + ' ...' )

  with open( fName, 'r' ) as fp :
    lines = fp.read().replace( '\r', '' ).split( '\n' )

  for line in lines :
    for token in line.split() :
      processToken( token )

#--------------------------------------------------
if ( __name__ == '__main__' ) :
  main()

#--------------------------------------------------

File: HMWK_02_tvt3804/hmwk_02.cs
// Tran, Thang
// 1001-233-804
// 2018-03-10

using System;
using System.IO;
using System.Text.RegularExpressions;

public class hmwk_02
{
    static public void processToken(string token)
    {
        //Defile Regular Expresion 
        Regex isID = new Regex(@"^[a-mN-Z_][n-zA-M0-9_]*$");
        Regex isInt = new Regex(@"^(0d|0D)[0-9]+$");
        Regex isnotEFP = new Regex(@"^[0-9]+$");
        Regex isHexInt = new Regex(@"^(0x|0X)[0-9a-fA-F]+$");
        Regex isFP = new Regex(@"^[0-9]+\.[0-9]+$");
        Regex isEFP = new Regex(@"^(([0-9]+\.)|(\.[0-9]+)|([0-9]+\.[0-9]+)|([0-9]+))([eE][-+]?[0-9]+)?$");

        //Condition to match with RE
        if (isID.Match(token).Success)
        {
            Console.WriteLine(">" + token + "< matches ID.");
        }
        else if (isInt.Match(token).Success)
        {
            Console.WriteLine(">" + token + "< matches INT.");
        }
        else if (isHexInt.Match(token).Success)
        {
            Console.WriteLine(">" + token + "<  matches HEXINT.");
        }
        else if (isFP.Match(token).Success)
        {
            Console.WriteLine(">" + token + "<  matches FP.");
        }
        else if (isnotEFP.Match(token).Success)
        {
            Console.WriteLine(">" + token + "<  does not match.");
        }
        else if (isEFP.Match(token).Success)
        {
            Console.WriteLine(">" + token + "<  matches EFP.");
        }
        else
        {
            Console.WriteLine(">" + token + "<  does not match.");
        }
    }

    static public void Main(string[] args)
    {
        Console.WriteLine("processing tokens from " + args[0] + " ...");

        foreach (string line in File.ReadAllLines(args[0]))
        {
            foreach (string token in line.Split((char[])null, StringSplitOptions.RemoveEmptyEntries))
            {
                processToken(token);
            }
        }
    }
}
